source/common/network/connection_impl.cc

```diff
@@ -284,9 +284,19 @@ void ConnectionImpl::setBufferLimits(uint32_t limit) {
   // asynchronously, we have the option of either setting the watermarks aggressively, and regularly
   // enabling/disabling reads from the socket, or allowing more data, but then not triggering
   // based on watermarks until 2x the data is buffered in the common case.  Given these are all soft
-  // limits we err on the side of buffering more and having better performace.
+  // limits we err on the side of buffering more triggering watermark callbacks less often.
```

Comment:
Heh, just had the same conversation with Harvey.   It is the common case for tcp proxying.  I'd updated the comment to this effect but yeah, I'm just as happy removing.